src/typesAndDefaults.ts

```diff
@@ -54,17 +54,24 @@ export interface OptionsInterface<C extends Context, RT extends RedisType> {
 
   /**
    * @param ctx Is the context object you get from grammy/telegraf.
-   * @returns A number in **string** format as the unique key (identifier).
-   * @description A function to generate a unique key for every user. You cound set it as any key you want (e.g group id)
+   * @returns A unique **string** key (identifier).
+   * @description A function to generate a unique key for every user. You could set it as any key you want (e.g. group id)
    * @see [Getting Started](https://github.com/Amir-Zouerami/rateLimiter#-how-to-use)
    */
   keyGenerator?: (ctx: C) => string | undefined;
+
+  /**
+   * @default "RATE_LIMITER"
+   * @description A string prefix that is getting added to the storage key after calling the `keyGenerator()`.
+   */
+  keyPrefix?: string | undefined;
 }
 
-export const defaultOptions = {
+export const defaultOptions: OptionsInterface<Context, RedisType> = {
   timeFrame: 1000,
   limit: 1,
   onLimitExceeded: (_ctx: Context, _next: NextFunction) => {},
   storageClient: "MEMORY_STORE",
-  keyGenerator: (ctx: Context) => ctx.from === undefined ? undefined : ctx.from.id.toString(),
+  keyGenerator: (ctx: Context) => ctx.from?.id.toString(),
```

Comment:
```suggestion
  keyGenerator: (ctx: Context) => ctx.from?.id?.toString(),
```

In case `ctx.from` is null, the `id.toString()` part would also cause an error. You should make that optional as well if you wanna write it like this.

Reply:
It was just a rewrite of your code, it should have the same behavior.

Also, according to [this](https://github.com/grammyjs/ratelimiter/blob/main/src/typesAndDefaults.ts#L2), the ID can't be `undefined`.

BTW, why aren't you importing these types from the grammY?

Reply:
The last time I looked at this code was 3 years ago :)))  You're right. I thought about the id of a user being undefined which is not possible.

I did not import the types from grammy because rateLimiter supports both grammy and telegraf frameworks. I don't think it would cause any problems to import them from grammy but it would be depending on a specific framework and in case grammy decides to do something fancy with its types, I don't want this to be dependent on grammy in any way.